rectangle.h

```diff
@@ -1,31 +1,37 @@
 #pragma once
 
-#include <glm/glm.hpp>
-
 #include "indexbuffer.h"
 #include "renderer.h"
 #include "shader.h"
 #include "texture.h"
 #include "vertexarray.h"
 #include "vertexbuffer.h"
 
+#include <glm/glm.hpp>
+
+#include <memory>
+
+static const unsigned int kIndicies[] = {
```

Comment:
prefer std::array. Lots of benefits. You can get the c array out of it with .data() I believe